Tests/DependenciesTests/DependencyValuesTests.swift

```diff
@@ -71,6 +71,43 @@ final class DependencyValuesTests: XCTestCase {
     }
   }
 
+  func testOptionalDependency() {
+    for value in [nil, ""] {
+      withDependencies {
+        $0.optionalDependency = value
+      } operation: {
+        @Dependency(\.optionalDependency) var optionalDependency: String?
+        XCTAssertEqual(optionalDependency, value)
+      }
+    }
+  }
+
+  func testOptionalDependencyLive() {
+    withDependencies {
+      $0.context = .live
+    } operation: {
+      @Dependency(\.optionalDependency) var optionalDependency: String?
+      XCTAssertEqual(optionalDependency, "live")
+    }
+
+    withDependencies {
+      $0.context = .live
+      $0.optionalDependency = nil
+    } operation: {
+      @Dependency(\.optionalDependency) var optionalDependency: String?
+      XCTAssertNil(optionalDependency)
+    }
+  }
+
+  func testOptionalDependencyUndefined() {
+    @Dependency(\.optionalDependency) var optionalDependency: String?
+    XCTExpectFailure {
+      XCTAssertNil(optionalDependency)
+    } issueMatcher: {
+      $0.compactDescription.contains(#"Unimplemented: @Dependency(\.optionalDependency) …"#)
+    }
+  }
```

Comment:
Core libs XCTest unfortunately doesn't support `XCTExpectFailure`.

```suggestion
  #if !os(Linux)
  func testOptionalDependencyUndefined() {
    @Dependency(\.optionalDependency) var optionalDependency: String?
    XCTExpectFailure {
      XCTAssertNil(optionalDependency)
    } issueMatcher: {
      $0.compactDescription.contains(#"Unimplemented: @Dependency(\.optionalDependency) …"#)
    }
  }
  #endif
```